assembly_finder/rules/assembly_table.py

```diff
@@ -168,22 +169,29 @@ def select_assemblies(self, table):
         fact_table = table.replace({'RefSeq_category': {'reference genome': 0, 'representative genome': 1, 'na': 6},
                                     'AssemblyStatus': {'Complete Genome': 2, 'Chromosome': 3, 'Scaffold': 4,
                                                        'Contig': 5, 'na': 6}})
+        # make sure the path to Genbank ftp is available
+        fact_table = fact_table[fact_table.FtpPath_GenBank != '']
         sorted_table = fact_table.sort_values(['RefSeq_category', 'AssemblyStatus', 'Contig_count',
                                                'ScaffoldN50', 'ContigN50', 'AsmReleaseDate_GenBank'],
                                               ascending=[True, True, True, False, False, False])
+        
         if self.rank_to_select != 'None':
             logging.info(f'Filtering according to {self.rank_to_select}, Refseq categories, assembly status, '
                          f'contig count and release date')
             select_index = []
             unique_list = list(set(sorted_table[self.rank_to_select]))
-            if len(unique_list) > 1:
+            if len(unique_list) > 0:
                 for i in unique_list:
-                    select_index.append(sorted_table[sorted_table[self.rank_to_select] == i].sample(1).index[0])
-                    # randomly select one assembly ID for each unique selected rank (species for example)
+                    target = sorted_table[sorted_table[self.rank_to_select] == i]
+                    if len(target) >= self.n_by_rank:
+                        select_index += target.sample(self.n_by_rank).index.to_list()
```

Comment:
should be 
`target.iloc[0:self.n_by_rank].index.to_list()`
instead of 
`target.sample(self.n_by_rank).index.to_list()`